src/commands/run.rs

```diff
@@ -113,8 +114,36 @@ impl RunCommand {
         }
 
         // Load the main wasm module.
-        self.handle_module(&store, &module_registry)
-            .with_context(|| format!("failed to run main module `{}`", self.module.display()))?;
+        match self
+            .handle_module(&store, &module_registry)
+            .with_context(|| format!("failed to run main module `{}`", self.module.display()))
+        {
+            Ok(()) => (),
+            Err(e) => {
+                // If the program exited because of a trap, return an error code
+                // to the outside environment indicating a more severe problem
+                // than a simple failure.
+                for cause in e.chain() {
+                    if cause.is::<Trap>() {
```

Comment:
FWIW when using anyhow I think we can just do `e.downcast_ref::<Trap>()` and it'll automatically check the entire chain for `Trap` without having to use `chain()` explicitly

Reply:
Cool, and we can even simplify it to just `.is::<Trap>()` here.